### PR TITLE
Fix the Not Fully Evolved validator rule for past gens

### PR DESCRIPTION
**Pull request by KrisXV**

### CODE REVIEW

**Comment by KrisXV** [comment]
bump @TheImmortal

File: data/rulesets.ts [hunk 1]
```diff
@@ -649,7 +649,8 @@ export const BattleFormats: {[k: string]: FormatsData} = {
 		desc: "Bans Pok&eacute;mon that are fully evolved or can't evolve",
 		onValidateSet(set) {
 			const species = this.dex.getSpecies(set.species);
-			if (!species.nfe) {
+			const feInCurrentGen = species.evos && this.dex.getSpecies(species.evos[0]).gen > this.gen;
+			if (!species.nfe || feInCurrentGen) {
```
Comment (Zarel) [inline data/rulesets.ts:653]: It seems to me that the correct answer here is to set `.nfe` differently depending on the generation.

Comment (Zarel) [reply data/rulesets.ts:653]: (i.e. this implementation goes in `dex-data.ts` or `dex.ts` rather than here)

Comment (KrisXV) [reply data/rulesets.ts:653]: @Zarel how would you approach this? you can't currently access any of the functions in `dex.ts` in `dex-data.ts` and it makes more sense to handle `this.nfe` in `dex-data.ts`

Comment (Zarel) [reply data/rulesets.ts:653]: I would just give up and do it in `getSpecies`.

**Comment by KrisXV** [comment]
I'll reopen this later

**Pull request closed without merging**